libraries/SITL/SIM_Sprayer.h

```diff
@@ -0,0 +1,64 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  simple sprayer simulation class
+*/
+
+#pragma once
+
+#include "SIM_Aircraft.h"
+
+namespace SITL {
+
+class Sprayer {
+public:
+    const uint8_t pump_servo;
+    const int8_t spinner_servo;
+
+    Sprayer(const uint8_t _pump_servo, int8_t _spinner_servo) :
+        pump_servo(_pump_servo),
+        spinner_servo(_spinner_servo)
+    {}
+
+    // update sprayer state
+    void update(const struct Aircraft::sitl_input &input);
+
+    float payload_mass() const { return capacity; }; // kg; assume water
+
+private:
+
+    const uint32_t report_interval = 1000000; // microseconds
+    uint64_t last_report_us;
+
+    // const float pump_max_rate = 0.1; // litres/second
```

Comment:
Yeah.  For debugging purposes it is useful to have more in the tank and higher flow rate.

Worked well on the virtual ground, but it was probably asking a little much for the little multicopter to take off with 10kg.  I'll remove the commented lines.